crd-generator/api/src/main/java/io/fabric8/crd/generator/CRDGenerator.java

```diff
@@ -105,7 +120,10 @@ Map<String, AbstractCustomResourceHandler> getHandlers() {
     return handlers;
   }
 
-  public CRDGenerator customResourceClasses(Class<? extends CustomResource>... crClasses) {
+  // this is public API, so we cannot change the signature, so there is no way to prevent the possible heap pollution
```

Comment:
Not sure I understand the purpose fo this comment…

Reply:
IntelliJ showed a warning for this method regarding heap pollution ([this blog article explains it quite well](https://arbitrary-but-fixed.net/teaching/java/2018/02/01/java-generic-arrays-varargs.html)). It comes down to the combination of generics and varargs.

I assume this is public API and the method signature cannot be changed, so there is nothing we can do to prevent the possible heap pollution. To get rid of the warning we could either:
1. Make the method `final` and annotate with `@SafeVarargs`.
1. Add a `@SuppressWarnings("unchecked")` annotation.

I wasn't sure if 1. is an option, since it would also change the method signature and may break custom `CRDGenerator` implementations. So I chose 2. as solution.

Reply:
Thanks! Wasn't aware of this issue.